extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java

```diff
@@ -96,13 +102,31 @@ public void getSwaggerUiFinalDestination(
 
             }
 
+            if (devServicesLauncherConfig.isPresent()) {
+                DevServicesLauncherConfigResultBuildItem devServicesLauncherConfigResult = devServicesLauncherConfig.get();
+                Map<String, String> devServiceConfig = devServicesLauncherConfigResult.getConfig();
+                if (devServiceConfig != null && !devServiceConfig.isEmpty()) {
+                    // Map client Id from OIDC Dev Services
+                    if (devServiceConfig.containsKey(OIDC_CLIENT_ID) && !swaggerUiConfig.oauthClientId.isPresent()) {
+                        String clientId = devServiceConfig.get(OIDC_CLIENT_ID);
+                        swaggerUiConfig.oauthClientId = Optional.of(clientId);
+                    }
+                }
+            }
+
             String openApiPath = nonApplicationRootPathBuildItem.resolvePath(openapi.path);
             String swaggerUiPath = nonApplicationRootPathBuildItem.resolvePath(swaggerUiConfig.path);
 
             AppArtifact artifact = WebJarUtil.getAppArtifact(curateOutcomeBuildItem, SWAGGER_UI_WEBJAR_GROUP_ID,
                     SWAGGER_UI_WEBJAR_ARTIFACT_ID);
 
             if (launchMode.getLaunchMode().isDevOrTest()) {
+
+                // In dev mode, default to persist Authorization true
+                if (!swaggerUiConfig.persistAuthorization.isPresent()) {
+                    swaggerUiConfig.persistAuthorization = Optional.of(true);
+                }
```

Comment:
Instead of making it an `Optional`, isn't easier to define as a `boolean` with `true`  as the default value?

Reply:
The reason for this is that there is actually no default. In dev mode we make it `true`, except if explicitly set `false` by the user. In other modes it will be `false`. except if explicitly set `true` by the user

Reply:
My only concern is that it's not common to change config values in a processor, so perhaps this could be tested when this value is consumed

Reply:
It gets consumed in SmallRye, and there we do not have the concept of dev mode...